### PR TITLE
Fix setup for linux OSs

### PR DESCRIPTION
**Pull request by sirfoga**
**Fixes**
- problems when running `pip` (or `pip3`) or even (`python3 setup.py install`)
- issues when joining API urls (run the script and always got `404` errors)

**Extra**
- refactor main
- prettify code

### CODE REVIEW

File: turtl_backup.py [hunk 1]
```diff
@@ -90,6 +90,12 @@ def build_basic_auth(auth_token):
     return 'Basic ' + b64encode(b'user:' + auth_token).decode()
 
 
+def hide_password(password):
```
Comment (JulienPalard) [inline turtl_backup.py]: I really don't like this, can you please just not print the password at all ? Not even `***`, no even `"and password:"`, just no password?

**Review by JulienPalard — CHANGES_REQUESTED** [review]
Hi and thanks for all the work!

I was experimenting with cx_Freeze and it was working (at least on my machine with a specific version of cffi and a specific version of cryptography).

I'm OK to drop cxfreeze and I'm almost OK with all your changes. I'm however not OK with printing the password, even a few chars of it, even just its length, just nothing.

File: requirements.txt [hunk 2]
```diff
@@ -1,3 +1,4 @@
+requests
```
Comment (JulienPalard) [inline requirements.txt]: Could you please pin a specific version of requests here? a requirements.txt is aimed to be extremly specific, so anyone could reproduce an exact environment while working (contrary to a setup.py install_requirements which should be as broad as possible to keep compatibility with other installed packages).

File: turtl_backup/turtl_backup_gui.py [hunk 3]
```diff
@@ -1,7 +1,13 @@
 #!/usr/bin/env python3
 
 import sys
+
```
Comment (JulienPalard) [inline turtl_backup/turtl_backup_gui.py]: As you dropped cxFreeze, this file is now useless, please drop it completly (and hook the entry point directly to the main of gui.py).

File: turtl_backup.py (hunk 1 above)
Comment (sirfoga) [reply turtl_backup.py]: fix in 7a469f4dd16d82766c7697b0d5308bd63c910290
removed user print in 55d776fbaef23ea34e9898da182ef288939f8ba5

File: requirements.txt (hunk 2 above)
Comment (sirfoga) [reply requirements.txt]: fix d05a5e66b1eae5c390ce58ecc96f8916bf1c51e2

File: turtl_backup/turtl_backup_gui.py (hunk 3 above)
Comment (sirfoga) [reply turtl_backup/turtl_backup_gui.py]: fix 91ca6fa9a16723fef9624d27d1556e4f5777610b

File: setup.py [hunk 4]
```diff
@@ -35,8 +15,7 @@
     url='https://github.com/JulienPalard/turtle-backup',
     entry_points={
         'console_scripts': [
-            'turtl-backup=turtl_backup.turtl_backup:main',
-            'turtl-backup-gui=turtl_backup.gui:main'
```
Comment (JulienPalard) [inline setup.py]: Why removing the `console_scripts` `turtl-backup-gui`?

(this was not correlated with the `turtl-backup-gui.py` which served only as a wrapper for cxFreeze)

File: turtl_backup/gui.py [hunk 5]
```diff
@@ -2,25 +2,26 @@
...
 import tkinter as tk
 import tkinter.filedialog as tkFileDialog
 import tkinter.messagebox as tkMessageBox
+from datetime import date
 from urllib.parse import urljoin
 
 import requests
-from turtl_backup import turtl_backup
+
+import turtl_backup
```
Comment (JulienPalard) [inline turtl_backup/gui.py]: I think you broke the gui here, there is probably no `build_basic_auth` in the turtl_backup package, but maybe in the module?

File: setup.py (hunk 4 above)
Comment (sirfoga) [reply setup.py]: yes, you're right, fix in b52a047f03456d55a91c718c0091b94ac302d706

File: turtl_backup/gui.py (hunk 5 above)
Comment (sirfoga) [reply turtl_backup/gui.py]: fix in 1b9ac241ccf78e5e324c3157967b7ca407b694fa

**Comment by sirfoga** [comment]
I've added `fetch_backup` to allow code re-use when fetching the backup response

File: turtl_backup/gui.py [hunk 6]
```diff
@@ -52,11 +52,8 @@ def save(self):
         username = self.entry_user.get()
         password = self.entry_password.get()
         auth = turtl_backup.get_auth(username, password)
-        basic_auth = turtl_backup.build_basic_auth(auth)
         try:
-            response = requests.get(urljoin(server, '/sync/full'),
-                                    headers={'authorization': basic_auth})
-            response.raise_for_status()
```
Comment (JulienPalard) [inline turtl_backup/gui.py]: By removing this line, the GUI will never show the error (if an error occur while downlowding the file) and will prompt the save_as, probably crashing right after at the ` response.text` line.

You should not modify tons of things in a single PR, you started by just removing cxFreeze from the setup.py, and ended refactoring everything. I would have merged sooner if the changes were simplier, and you're still able to open multiple distincts PR for multiple unrelated modifications.

I will still merge this huge one, and thanks for all the work, once it looks OK (hint: stop modifying things on this PR, just fix my reviews).

**Review by JulienPalard — CHANGES_REQUESTED** [review]

File: turtl_backup/gui.py (hunk 6 above)
Comment (sirfoga) [reply turtl_backup/gui.py]: restored with 900176c8718a7a033a594d75b6168dcb80e2cbc3

**Comment by JulienPalard** [comment]
Thanks! Next time (on any project) try to keep your PR more atomic, a PR titled "Fix setup" is not expected to refactor everything, it's hard to review, takes time to merge, and each time you add a new commit to it, you risk introducing a new problem, postponing the merge.

In other words, the expected workflow is "PR, review, fix, merge", your workflow was "PR, review, touch something unrelated while fixing, review, touch something unrelated while fixing, review, touch something unrelated while fixing, review, fix without touching unrelated, merge".

**Comment by sirfoga** [comment]
Yes, I'm sorry for the trouble! Thank anyway for the project!!

**Pull request merged**